connexion/decorators/parameter.py

```diff
@@ -49,20 +51,27 @@ def wrapper(*args, **kwargs):
         logger.debug('Function Arguments: %s', arguments)
 
         try:
-            body_parameters = flask.request.json or {}  # type: dict
+            body_parameters = flask.request.json or {}
         except exceptions.BadRequest:
             body_parameters = {}
 
         # Add body parameters
-        for key, value in body_parameters.items():
-            if key not in arguments:
-                logger.debug("Body parameter '%s' not in function arguments", key)
+        if isinstance(body_parameters, dict):
```

Comment:
I think getting the value from the body parsed as dict is wrong. 'There can be one "body" parameter at most.' the Swagger spec says.